src/main/java/banking/BankingSystem.java

```diff
@@ -0,0 +1,177 @@
+            case 2:
+                System.out.println("Enter income:");
+                int amount = scanner.nextInt();
+                accountDAO.addBalance(currentAccount.getId(), amount);
+                break;
+            case 3:
+                transferMoney();
+                break;
+            case 4:
+                accountDAO.deleteAccount(currentAccount.getId());
+                System.out.println("The account has been closed!");
+                isLogin = false;
+                currentAccount = null;
+                break;
+            case 5:
+                break;
+            case 0:
+                System.out.println("Bye!");
+                exit(0);
+            default:
+                System.out.println("Wrong number");
+                break;
+        }
+    }
+
+    private void transferMoney() {
+        System.out.println("Enter card number:");
+        String card = scanner.next();
+        if (!card.equals(accountDAO.getNumber(currentAccount.getId()))) {
+            checkValid(card);
+        } else {
+            System.out.println("You can't transfer money to the same account!");
+        }
+    }
+
+    private void checkValid(String card) {
+        if (cardValidator.isValidCard(this.cardGenerator, card)) {
+            enterMoneyCount(card);
+        } else {
+            System.out.println("Probably you made a mistake in the card number. Please try again!");
+        }
+    }
+
+    private void enterMoneyCount(String card) {
+        if (accountDAO.containCard(card)) {
+            System.out.println("Enter how much money you want to transfer:");
+            amountSend = scanner.nextInt();
+            transaction(card);
+        } else {
+            System.out.println("Such a card does not exist.");
+        }
+    }
+
+    private void transaction(String card) {
+        if (amountSend <= accountDAO.getBalance(currentAccount.getId())) {
+            accountDAO.updateMoney(amountSend, card);
+            accountDAO.updateMoney(-amountSend, currentAccount.getCard().getNumber());
+            System.out.println("success");
+        } else {
+            System.out.println("Not enough money!");
+        }
+    }
+
+    private void logInOption() {
+        System.out.println("Enter your card number:");
+        String number = scanner.next();
+
+        System.out.println("Enter your PIN:");
+        String PIN = scanner.next();
+
+        Account account = accountDAO.login(number, PIN);
+        if (account == null) {
+            System.out.println("Wrong card number or PIN");
+        } else {
+            System.out.println("You have successfully logged in");
+            isLogin = true;
+            currentAccount = account;
+        }
+    }
+
+    private void createAccount() {
+        Card card = cardGenerator.generate();
+        Account account = new Account(
+                accountDAO.getId() + 1,
+                card,
+                0);
+        accountDAO.insertCard(account);
```

Comment:
You never use `id` in the `insertCard` method. Do you need it?

Reply:
i use getId from Account for check balance, transfer money and delete account in Banking System

Reply:
The question asks about `AccountDAO`. insertCard(...). 
Please check the question one more time.

Reply:
I changed this method, check please in new version